### ISSUE TITLE
Cannot get executing test method in `ParameterResolver` of `@MethodSource` factory method

### ISSUE THREAD

**Issue by pxzxj** [junit-team/junit5#3323]
## Deliverables

- [ ] Get test method in `ParameterResolver` of parameterizedTest factory method. for example below `extensionContext.getTestMethod()` should return method `testWithFactoryMethodWithArguments`

```java
public class MyTest {

	@RegisterExtension
	static final IntegerResolver integerResolver = new IntegerResolver();

	@ParameterizedTest
	@MethodSource("factoryMethodWithArguments")
	void testWithFactoryMethodWithArguments(String argument) {
		assertTrue(argument.startsWith("2"));
	}

	static Stream<Arguments> factoryMethodWithArguments(int quantity) {
		return Stream.of(
				arguments(quantity + " apples"),
				arguments(quantity + " lemons")
		);
	}

	static class IntegerResolver implements ParameterResolver {

		@Override
		public boolean supportsParameter(ParameterContext parameterContext,
										 ExtensionContext extensionContext) {
			return parameterContext.getParameter().getType() == int.class;
		}

		@Override
		public Object resolveParameter(ParameterContext parameterContext,
									   ExtensionContext extensionContext) {
			Optional<Method> testMethod = extensionContext.getTestMethod();
			boolean present = testMethod.isPresent();  //false
			return 2;
		}

	}
}

```

## Related issues

- #3445

**Comment by marcphilipp** [comment]
Accessing the method from the `ParameterContext` as mentioned below should work.

https://github.com/junit-team/junit5/blob/49e74f755d8a1c9c6a341b7220ab6f72a5c4cfa9/junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/ParameterResolver.java#L77-L79

**Comment by pxzxj** [comment]
@marcphilipp actually `parameterContext.getDeclaringExecutable()` return the factory method, not test method

![factory-method](https://github.com/junit-team/junit5/assets/28888745/ef7c4b65-8824-4296-bd0a-3be6b4636068)

**Comment by marcphilipp** [comment]
@pxzxj Thanks for the feedback! You're correct. This use case is not currently supported.

**Team decision:** Investigate whether it's feasible to support this use case.

**Comment by JojOatXGME** [comment]
I took a short look at this issue while preparing #4032. I was not able to reproduce the issue descriped here on the current `main` branch (d0130853639346c962609e9f8b711336879e39ca). I copied the code from the description and executed it. For me, `ExtensionContext.getTestMethod()` did return the test method `testWithFactoryMethodWithArguments` in `IntegerResolver.resolveParameter`.

**Comment by juanplopes** [comment]
I've encountered this issue after having the same problem with version `5.10.0`. Reading @JojOatXGME's comment, I took a chance and upgraded to `5.11.1` and, indeed, the problem does not happen on that version.

**Comment by marcphilipp** [comment]
This was indeed fixed in #3905.

@JojOatXGME Thanks for trying to reproduce it! :+1:
